Source/WebCore/dom/Document.cpp

```diff
@@ -8462,6 +8462,13 @@ void Document::cancelIdleCallback(int id)
     m_idleCallbackController->removeIdleCallback(id);
 }
 
+bool Document::hasPendingIdleCallback() const
+{
+    if (activeDOMObjectsAreSuspended() || activeDOMObjectsAreStopped())
+        return false;
+    return m_idleCallbackController ? !m_idleCallbackController->isEmpty() : false;
```

Comment:
Nit - _maybe_ reads slightly cleaner this way?

```suggestion
    return m_idleCallbackController && !m_idleCallbackController->isEmpty();
```

Reply:
Sure.